### PR TITLE
add options to draw control

### PR DESCRIPTION
**Pull request by EtsuNDmA**
Add options for [draw control](http://leaflet.github.io/Leaflet.draw/docs/leaflet-draw-latest.html#options). Also add custom filename for geojson file

### CODE REVIEW

File: folium/plugins/draw.py [hunk 1]
```diff
@@ -10,28 +13,36 @@
...
     ----------
     export : bool, default False
         Add a small button that exports the drawn shapes as a geojson file.
-
+    filename : string, default data.geojson
+        Name of geojson file
+    position : string, default 'topleft'
+        Position of control. It can be one of 'topleft', 'toprigth', 'bottomleft', 'bottomright'
+    draw : dict, default None
+        The options used to configure the draw toolbar
```
Comment (Conengmo) [inline folium/plugins/draw.py]: 1. Can you link to a page where these options are listed? Same for `edit`.
2. I'm not sure about the names `draw` and `edit`, they are too general. The class is named `Draw`, so an object is probably named `draw`. For these arguments, how about `draw_options` and `edit_options`?

File: folium/plugins/draw.py [hunk 2]
```diff
@@ -10,28 +13,36 @@
...
+    edit : dict, default None
+        The options used to configure the edit toolbar
     Examples
     --------
     >>> m = folium.Map()
     >>> Draw(export=True).add_to(m)
-
```
Comment (Conengmo) [inline folium/plugins/draw.py]: Why did you remove this whitespace?

File: folium/plugins/draw.py [hunk 3]
```diff
@@ -43,37 +54,48 @@ class Draw(MacroElement):
...
+              document.getElementById('export').setAttribute(
+                'download',
+                "{{kwargs.get('filename', 'data.geojson')}}"
+              );
+            }
             {% endmacro %}
             """)
 
-    def __init__(self, export=False):
+    js_url = 'https://cdnjs.cloudflare.com/ajax/libs/leaflet.draw/1.0.2/leaflet.draw.js'
```
Comment (Conengmo) [inline folium/plugins/draw.py]: Why did you move the URLs here? Throughout folium they are in the `render` method, so I'd prefer to keep them there for consistency.

File: folium/plugins/draw.py [hunk 4]
```diff
@@ -10,28 +13,36 @@
 class Draw(MacroElement):
     """
     Vector drawing and editing plugin for Leaflet.
-
     Parameters
     ----------
     export : bool, default False
         Add a small button that exports the drawn shapes as a geojson file.
-
+    filename : string, default data.geojson
```
Comment (Conengmo) [inline folium/plugins/draw.py]:
```suggestion
    filename : string, default 'data.geojson'
```

File: folium/plugins/draw.py [hunk 5]
```diff
@@ -10,28 +13,36 @@
...
+        Name of geojson file
+    position : string, default 'topleft'
+        Position of control. It can be one of 'topleft', 'toprigth', 'bottomleft', 'bottomright'
+    draw : dict, default None
```
Comment (Conengmo) [inline folium/plugins/draw.py]:
```suggestion
    draw : dict, optional
```

File: folium/plugins/draw.py [hunk 6]
```diff
@@ -10,28 +13,36 @@
...
+        The options used to configure the draw toolbar
+    edit : dict, default None
```
Comment (Conengmo) [inline folium/plugins/draw.py]:
```suggestion
    edit : dict, optional
```

File: folium/plugins/draw.py [hunk 7]
```diff
@@ -43,37 +54,48 @@ class Draw(MacroElement):
...
+        self.filename = filename
+        self.position = position
+        self.draw = draw or {}
+        self.edit = edit or {}
 
     def render(self, **kwargs):
-        super(Draw, self).render()
+        super(Draw, self).render(
+            position=self.position,
+            draw=json.dumps(self.draw),
```
Comment (Conengmo) [inline folium/plugins/draw.py]: These variables are already available in `self`, you don't have to pass them here. You can do the json dumping in `__init__`.

**Comment by EtsuNDmA** [comment]
Hi! @Conengmo, I took into account your comments. Also add an example to dockstring

**Comment by Conengmo** [comment]
Looks good @EtsuNDmA! I'll merge it after checking it one last time, but I don't think any more changes are necessary.

**Comment by Conengmo** [comment]
Thanks @EtsuNDmA!
